task8/src/java/Ad.java

```diff
@@ -0,0 +1,157 @@
+package com;
+
+import java.io.Serializable;
+import java.util.Date;
+
+public class Ad implements Serializable {
+    private String id;
+    private String description;
+    private Date createdAt;
+    private String link;
+    private String vendor;
+    private String photoLink;
+    private String[] hashTags;
+    private String discount;
+    private Date validUntil;
+    private int rating;
+    private String[] reviews;
+
+    public Ad(){}
+
+    public Ad(String id, String description, Date createdAt, String link, String vendor, String photoLink,
+              String[] hashTags, String discount, Date validUntil, int rating, String[] reviews) {
+        this.id = id;
+        this.description = description;
+        this.createdAt = createdAt;
+        this.link = link;
+        this.vendor = vendor;
+        this.photoLink = photoLink;
+        this.hashTags = hashTags;
+        this.discount = discount;
+        this.validUntil = validUntil;
+        this.rating = rating;
+        this.reviews = reviews;
+    }
+
+    public Ad(Ad another){
+        this.id = another.id;
+        this.description = another.description;
+        this.createdAt = another.createdAt;
+        this.link = another.link;
+        this.vendor = another.vendor;
+        this.photoLink = another.photoLink;
+        this.hashTags = another.hashTags;
+        this.discount = another.discount;
+        this.validUntil = another.validUntil;
+        this.rating = another.rating;
+        this.reviews = another.reviews;
+    }
+
+    public static boolean validate(Ad item) {
```

Comment:
By "encapsulation", I meant creating an instance method, i.e. the method that belongs to a particular entity (like in the phrase "This Ad is a valid add"), and not just moving the static method around)